Tests/XcodeprojTests/FunctionalTests.swift

```diff
@@ -147,8 +147,10 @@ func XCTAssertXcodeBuild(project: AbsolutePath, file: StaticString = #file, line
         if env["TOOLCHAINS"] == nil {
             env["TOOLCHAINS"] = "default"
         }
+        let schemeName = String(project.basename.dropLast(10)) + "-Package"
+        let buildDir = project.parentDirectory.appending(component: "build").asString
         try Process.checkNonZeroExit(
-            args: "xcodebuild", "-project", project.asString, "-alltargets", environment: env)
+            args: "xcodebuild", "-project", project.asString, "-scheme", schemeName, "-derivedDataPath", buildDir, environment: env)
```

Comment:
Should we put in all ones that are in edit mode then?

As for the linking think, we might be able to do a hack like set `LD` and/or `LDPLUSPLUS` to be `/usr/bin/true` for that one target.